sapi/fpm/tests/socket-uds-basic2.phpt

```diff
@@ -0,0 +1,48 @@
+--TEST--
+FPM: UDS connection, expand $pool and create socket directory
+--SKIPIF--
+<?php include "skipif.inc"; ?>
+--FILE--
+<?php
+
+require_once "tester.inc";
+
+$dir = __DIR__ . '/' . basename(__FILE__, '.php') . '-$pool';
+$sock = "$dir/fpm.sock";
+$dir = __DIR__ . '/' . basename(__FILE__, '.php') . '-default';
+$sock_exp = "$dir/fpm.sock";
+
+$cfg = <<<EOT
+[global]
+error_log = {{FILE:LOG}}
+[default]
+listen = $sock
+ping.path = /ping
+ping.response = pong
+pm = dynamic
+pm.max_children = 5
+pm.start_servers = 2
+pm.min_spare_servers = 1
+pm.max_spare_servers = 3
+EOT;
+
+$tester = new FPM\Tester($cfg);
+$tester->start();
+$tester->expectLogStartNotices();
+$tester->ping($sock_exp);
+$tester->terminate();
+$tester->expectLogTerminatingNotices();
+$tester->close();
+
+var_dump(rmdir($dir));
```

Comment:
This is the only thing that I don't really like. FPM creates a dir and doesn't clean up after itself. I understand that there might fail if there are some other files but still maybe worth to try it. It would probably have to save that info in global that it should attempt to clean it up.

Is this done by some other projects. For example does nginx or httpd create a dir for UDS and if so, do they clean it up on shutdown?

Reply:
Indeed, system creates the run directory on startup, and delete it on shutdown.

Reply:
So if systemd cleans it up after itself why should not be removed by FPM when it's created by FPM?

Reply:
Don't find an easy way to clean crated directories.
As they can be used by various sockets.. probably need to be tracked in a separate array, need to think about it.

Or perhaps forget about this initial idea (as perhaps it is not so useful, in container the directory will have to be created at build time so can be mounted at runtime, and probably most people use network port for containers...)

Reply:
Yeah the usefulness of the feature seems a bit questionable to be honest. I think people are usually using just a single pool in container and as you say, it's usually using network port. It's also quite usual to have a specific fpm.conf that is already modified to for example change logging and other things. It's not that hard to also change the path in case someone still uses UDS (e.g. when running it together with nginx in a single container and using some process manager like s6). Also not that hard to create a dir in Dockerfile...